src/server/routes/contract/events/getContractEventLogs.ts

```diff
@@ -0,0 +1,132 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { getContractEventLogsByBlockAndTopics } from "../../../../db/contractEventLogs/getContractEventLogs";
+import { isContractSubscribed } from "../../../../db/contractSubscriptions/getContractSubscriptions";
+import { createCustomError } from "../../../middleware/error";
+import {
+  contractParamSchema,
+  standardResponseSchema,
+} from "../../../schemas/sharedApiSchemas";
+import { getChainIdFromChain } from "../../../utils/chain";
+
+const requestQuerySchema = Type.Object({
+  fromBlock: Type.Number(),
+  toBlock: Type.Optional(Type.Number()),
+  topics: Type.Optional(Type.Array(Type.String())),
+});
+
+const responseSchema = Type.Object({
+  result: Type.Object({
+    logs: Type.Array(
+      Type.Object({
+        chainId: Type.Number(),
+        contractAddress: Type.String(),
+        blockNumber: Type.Number(),
+        transactionHash: Type.String(),
+        topics: Type.Array(Type.String()),
+        data: Type.String(),
+        eventName: Type.Optional(Type.String()),
+        decodedLog: Type.Any(),
+        timestamp: Type.Number(),
+        transactionIndex: Type.Number(),
+        logIndex: Type.Number(),
+      }),
+    ),
+    status: Type.String(),
+  }),
+});
+
+responseSchema.example = {
+  result: {
+    chain: "ethereum",
+    contractAddress: "0x....",
+    firstIndexedBlock: 100,
+    lastIndexedBlock: 200,
+    status: "success",
+  },
+};
+
+export async function getContractEventLogs(fastify: FastifyInstance) {
+  fastify.route<{
+    Params: Static<typeof contractParamSchema>;
+    Reply: Static<typeof responseSchema>;
+    Querystring: Static<typeof requestQuerySchema>;
+  }>({
+    method: "GET",
+    url: "/contract/:chain/:contractAddress/events/get-logs",
+    schema: {
+      summary: "Get contract event logs",
+      description: "Get event logs for a subscribed contract",
+      tags: ["Contract", "Index"],
+      operationId: "read",
+      params: contractParamSchema,
+      querystring: requestQuerySchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseSchema,
+      },
+    },
+    handler: async (request, reply) => {
+      const { chain, contractAddress } = request.params;
+      const { fromBlock, toBlock, topics } = request.query;
```

Comment:
we should also maybe have a limit on the number of blocks user can query at once, `toBlock - fromBlock <= 100` or something.